src/hyp3_isce2/insar_tops_burst.py

```diff
@@ -201,44 +201,78 @@ def insar_tops_single_burst(
+    reference: Iterable[str],
+    secondary: Iterable[str],
+    swaths: list = [1, 2, 3],
+    looks: str = '20x4',
+    apply_water_mask=False,
+    bucket: Optional[str] = None,
+    bucket_prefix: str = '',
+):
+    validate_bursts(reference, secondary)
+    reference_safe_path = burst2safe(reference)
+    reference_safe = reference_safe_path.name.split('.')[0]
+    secondary_safe_path = burst2safe(secondary)
+    secondary_safe = secondary_safe_path.name.split('.')[0]
+
+    range_looks, azimuth_looks = [int(value) for value in looks.split('x')]
+    swaths = list(set(int(granule.split('_')[2][2]) for granule in reference))
+    polarization = reference[0].split('_')[4]
+
+    log.info('Begin ISCE2 TopsApp run')
+    insar_tops_packaged(
+        reference=reference_safe,
+        secondary=secondary_safe,
+        swaths=swaths,
+        polarization=polarization,
+        azimuth_looks=azimuth_looks,
+        range_looks=range_looks,
+        apply_water_mask=apply_water_mask,
+        bucket=bucket,
+        bucket_prefix=bucket_prefix,
+    )
+    log.info('ISCE2 TopsApp run completed successfully')
+
+
 def main():
     """HyP3 entrypoint for the burst TOPS workflow"""
     parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
-    parser.add_argument('--bucket', help='AWS S3 bucket HyP3 for upload the final product(s)')
-    parser.add_argument('--bucket-prefix', default='', help='Add a bucket prefix to product(s)')
+    parser.add_argument('--reference', type=str.split, nargs='+', help='List of reference scenes"')
+    parser.add_argument('--secondary', type=str.split, nargs='+', help='List of secondary scenes"')
     parser.add_argument(
         '--looks', choices=['20x4', '10x2', '5x1'], default='20x4', help='Number of looks to take in range and azimuth'
     )
     parser.add_argument(
-        '--apply-water-mask',
-        type=string_is_true,
-        default=False,
-        help='Apply a water body mask before unwrapping.',
-    )
-    parser.add_argument(
-        'granules',
-        type=str.split,
-        nargs='+',
-        help='Reference and secondary scene names'
+        '--apply-water-mask', type=string_is_true, default=False, help='Apply a water body mask before unwrapping.'
     )
+    parser.add_argument('--bucket', help='AWS S3 bucket HyP3 for upload the final product(s)')
+    parser.add_argument('--bucket-prefix', default='', help='Add a bucket prefix to product(s)')
 
     args = parser.parse_args()
 
-    granules = [item for sublist in args.granules for item in sublist]
-    if len(granules) != 2:
-        parser.error('No more than two granules may be provided.')
-    if len(granules[0]) != len(granules[1]) != 1:
-        parser.error('Must include 1 reference and 1 secondary.')
+    references = [item for sublist in args.reference for item in sublist]
+    secondaries = [item for sublist in args.secondary for item in sublist]
```

Comment:
Out of curiosity, what happens if an empty list is passed with `--reference ' '` or `--secondary ' '`?

Reply:
Nevermind, that's prohibited by `nargs='+'`